### PR TITLE
Pin Ubuntu Focal base image tag

### PR DESCRIPTION
**Pull request by bbezak**

### CODE REVIEW

File: etc/kayobe/kolla/kolla-build.conf [hunk 1]
```diff
@@ -1,6 +1,9 @@
...
+{# Packages in the latest upstream Ubuntu base image can be ahead of our package repo #}
+{# snapshots, so pin to a specific tag. #}
+{# This tag should be updated when Ubuntu package repo snapshot versions are changed. #}
+{% if kolla_base_distro == 'ubuntu' %}
+base_tag = focal-20231003
+{# Similarly pinning to Rocky 9 minor version used in our repos #}
+{% elif kolla_base_distro == 'rocky' %}
 base_tag = 9.{{ stackhpc_pulp_repo_rocky_9_minor_version }}
 {% endif %}
-build_args = {{ kolla_build_args.items() | map('join', ':') | join(',') }}
```
Comment (markgoddard) [inline etc/kayobe/kolla/kolla-build.conf:6]: Looks like we lost this

Comment (bbezak) [reply etc/kayobe/kolla/kolla-build.conf:6]: yeah just noticed that as well :)

Comment (bbezak) [reply etc/kayobe/kolla/kolla-build.conf:6]: copy paste fail 👎

File: etc/kayobe/kolla/kolla-build.conf [hunk 2]
```diff
@@ -1,6 +1,9 @@
-[DEFAULT]
```
Comment (Alex-Welsh) [inline etc/kayobe/kolla/kolla-build.conf:1]: Missing heading?

Comment (bbezak) [reply etc/kayobe/kolla/kolla-build.conf:1]: done

File: etc/kayobe/kolla/kolla-build.conf (hunk 1 above)
Comment (bbezak) [reply etc/kayobe/kolla/kolla-build.conf:6]: done

**Comment by bbezak** [comment]
test here https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/7260925521

**Review by markgoddard — APPROVED** [review]

**Pull request merged**